source/AI.cpp

```diff
@@ -396,23 +497,28 @@ void AI::UpdateKeys(PlayerInfo &player, Command &activeCommands)
 	if(activeCommands.Has(Command::DEPLOY))
 		IssueDeploy(player);
 
+	// The gather command controls formation flying when combined with shift.
+	const bool shift = activeCommands.Has(Command::SHIFT);
+	if(shift && activeCommands.Has(Command::GATHER))
+		IssueFormationChange(player);
+
 	shared_ptr<Ship> target = flagship->GetTargetShip();
 	shared_ptr<Minable> targetAsteroid = flagship->GetTargetAsteroid();
 	Orders newOrders;
-	if(activeCommands.Has(Command::FIGHT) && target && !target->IsYours())
+	if(activeCommands.Has(Command::FIGHT) && target && !target->IsYours() && !shift)
```

Comment:
if the following all have !shift, could that be packed into a single check for !shift?

Reply:
Yes, but I'm afraid that it would be less readable than what we have now. I consider shift to actually be part of the command itself. (Since both are player inputs.)

Reply:
Makes sense